curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.english.md

```diff
@@ -0,0 +1,65 @@
+---
+id: 587d7dbb367417b2b2512be6
+title: Check For Mixed Grouping of Characters
+challengeType: 1
+---
+
+## Description
+<section id='description'>
+Sometimes we want to check for groups of characters using a Regular Expression, to achieve this task we use the round brackets <code>()</code>.
+If you want to find either <code>"Penguin"</code> or <code>"Pumpkin"</code> in a string, you could use the following Regular Expression: <code>/P(engu|umpk)in/g</code>
+Then you can test the code using the <code>.test()</code> method to find whether the requires character(s)/string(s) are in the test string or not.
+<blockquote>let testStr = "Pumpkin";<br>let testRegex = /P(engu|umpk)in/g;<br>testRegex.test(testStr);<br>// Returns true</blockquote>
+</section>
+
+## Instructions
+<section id='instructions'>
+Fix the RegEx such that it checks for either <code>Hello</code> or <code>Halo</code> in the string such that the RegEx is case sensitive. Make sure that the characters of code you write are as less as possible.
+Then fix the code so that the RegEx that you have created is checked in a string and either <code>true</code> or <code>false</code> is returned depending on whether the RegEx matches.
+</section>
+
+## Tests
+<section id='tests'>
+
+```yml
+tests:
+  - text: You should use <code>()</code> in the regex.
+    testString: assert(code.match(/\(.*\)/g), 'You should use <code>()</code> in the regex.');
+  - text: Your regex should check for the presence of either of <code>Hello</code> or <code>Halo</code> in the string.
+    testString: assert(code.match(/\/H\(((el\|a)|(a\|el))\)lo\/g/g), 'Your regex should check for the presence of either of <code>Hello</code> or <code>Halo</code> in the string.');
+  - text: You should use <code>.test()</code> to test the regex.
+    testString: assert(code.match(/myRegex.test\(\s*myString\s*\);?/), 'You should use <code>.test()</code> to test the regex.');
```

Comment:
This one can be simpler, meaning harder to break — just check for `test\(.*\)`. You might also notice that either version removes the need for the first test.